lib/livebook/runtime/erl_dist/runtime_server.ex

```diff
@@ -753,7 +753,7 @@ defmodule Livebook.Runtime.ErlDist.RuntimeServer do
   end
 
   def handle_call(:fetch_proxy_handler, _from, state) do
-    if pid = GenServer.whereis(Kino.Proxy) do
+    if pid = Livebook.Proxy.Handler.get_pid(Kino.Proxy, self()) do
```

Comment:
I don't think is `self()`. I think we need to pass the PID from the controller all the way up to here!